crates/gobley-uniffi-bindgen/src/lib.rs

```diff
@@ -12,14 +12,57 @@ use fs_err as fs;
 use uniffi_bindgen::{BindingGenerator, Component, ComponentInterface, GenerationSettings};
 
 mod gen_kotlin_multiplatform;
-use gen_kotlin_multiplatform::{generate_bindings, Config};
+use gen_kotlin_multiplatform::{generate_bindings, Config, ConfigKotlinTarget};
+
+pub struct KotlinBindingGenerator {
+    pub force_multiplatform: bool,
+}
+
+impl Default for KotlinBindingGenerator {
+    fn default() -> Self {
+        Self {
+            force_multiplatform: false,
+        }
+    }
+}
+
+impl KotlinBindingGenerator {
+    pub fn new() -> Self {
+        Self::default()
+    }
+    
+    pub fn with_multiplatform(mut self, enabled: bool) -> Self {
+        self.force_multiplatform = enabled;
+        self
+    }
+}
 
-pub struct KotlinBindingGenerator;
 impl BindingGenerator for KotlinBindingGenerator {
     type Config = Config;
 
     fn new_config(&self, root_toml: &toml::value::Value) -> Result<Self::Config> {
-        Ok(root_toml.clone().try_into()?)
+        // Support both [bindings.kotlin] format and flat format for compatibility
+        let kotlin_config = root_toml
+            .get("bindings")
+            .and_then(|b| b.get("kotlin"))
+            .unwrap_or(root_toml);
+        
+        let mut config: Config = kotlin_config.clone().try_into()?;
+        
+        // Override with CLI flag if provided
+        if self.force_multiplatform {
+            config.kotlin_multiplatform = true;
+            // Ensure we have default targets if none specified
+            if config.kotlin_targets.is_empty() {
+                config.kotlin_targets = vec![
+                    ConfigKotlinTarget::Jvm,
+                    ConfigKotlinTarget::Android,
+                    ConfigKotlinTarget::Native,
```

Comment:
Maybe redundant? `Config::kotlin_targets()` already checks whether the `kotlin_targets` field is empty or not.